sklearn/preprocessing/_encoders.py

```diff
@@ -1238,6 +1360,12 @@ class OrdinalEncoder(OneToOneFeatureMixin, _BaseEncoder):
         "encoded_missing_value": [Integral, type(np.nan)],
         "handle_unknown": [StrOptions({"error", "use_encoded_value"})],
         "unknown_value": [Integral, type(np.nan), None],
+        "max_categories": [Interval(Integral, 1, None, closed="left"), None],
```

Comment:
An honest doubt: would it be possible to also accept a list of shape (n_features,) to adapt `max_categories` feature-wise?

I am thinking that would make models more flexible though harder to tune.

Reply:
Yea, I think it's worth accepting a list for `max_categories` or even a dictionary. The dictionary would map from feature name to `max_categories` for that feature.

In any case, I prefer doing it in a follow up PR to keep the scope of this PR smaller.

Reply:
https://github.com/scikit-learn/scikit-learn/issues/26013